tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js

```diff
@@ -48,485 +59,411 @@ const objects = [
 
 describe('Listing corner cases tests', () => {
     let s3;
-    before(done => {
+    before(async () => {
         const config = getConfig('default', { signatureVersion: 'v4' });
-        s3 = new AWS.S3(config);
-        s3.createBucket(
-            { Bucket }, (err, data) => {
-                if (err) {
-                    done(err, data);
-                }
-                async.each(
-                    objects, (o, next) => {
-                        s3.putObject(o, (err, data) => {
-                            next(err, data);
-                        });
-                    }, done);
-            });
+        s3 = new S3Client(config);
+        await s3.send(new CreateBucketCommand({ Bucket }));
+        await Promise.all(objects.map(o => s3.send(new PutObjectCommand(o))));
     });
-    after(done => {
-        s3.listObjects({ Bucket }, (err, data) => {
-            async.each(data.Contents, (o, next) => {
-                s3.deleteObject({ Bucket, Key: o.Key }, next);
-            }, () => {
-                s3.deleteBucket({ Bucket }, done);
-            });
-        });
+    after(async () => {
+        const data = await s3.send(new ListObjectsCommand({ Bucket }));
+        await Promise.all(data.Contents.map(o => s3.send(new DeleteObjectCommand({ Bucket, Key: o.Key }))));
+        await s3.send(new DeleteBucketCommand({ Bucket }));
     });
-    it('should list everything', done => {
-        s3.listObjects({ Bucket }, (err, data) => {
-            assert.strictEqual(err, null);
-            cutAttributes(data);
-            assert.deepStrictEqual(data, {
-                IsTruncated: false,
-                Marker: '',
-                Contents: [
-                    objects[0].Key,
-                    objects[1].Key,
-                    objects[2].Key,
-                    objects[3].Key,
-                    objects[4].Key,
-                    objects[5].Key,
-                    objects[6].Key,
-                    objects[7].Key,
-                    objects[8].Key,
-                    objects[9].Key,
-                ],
-                Name: Bucket,
-                Prefix: '',
-                MaxKeys: 1000,
-                CommonPrefixes: [],
-            });
-            done();
+    it('should list everything', async () => {
+        const { $metadata, ...data } = await s3.send(new ListObjectsCommand({ Bucket }));
+        cutAttributes(data);
+        assert.deepStrictEqual(data, {
+            Contents: [
+                objects[0].Key,
```

Comment:
Not related to your changes, but you should be able to do a `Contents: [...objects.map(obj => obj.key)],